deploy-service/common/src/main/java/com/pinterest/deployservice/scm/SourceControlManagerProxy.java

```diff
@@ -31,7 +33,21 @@ private SourceControlManager getSourceControlManager(String scmType) throws Exce
         return manager;
     }
 
-    public String getType() {
+    public List<String> getSCMs() throws Exception {
+        validSCMs = new ArrayList<String> (this.managers.keySet());
+        return validSCMs;
+    }
+
+    public Boolean approved(String scmName) {
+        for (String validSCM : validSCMs) {
```

Comment:
try to use a function designed to check membership, e.g. https://stackoverflow.com/questions/37330551/how-to-check-whether-a-liststring-contains-a-specific-string